sdks/java/core/src/main/java/org/apache/beam/sdk/io/AvroSource.java

```diff
@@ -393,7 +393,7 @@ private static InputStream decodeAsInputStream(byte[] data, String codec) throws
       ByteArrayInputStream byteStream = new ByteArrayInputStream(data);
       switch (codec) {
         case DataFileConstants.SNAPPY_CODEC:
-          return new SnappyCompressorInputStream(byteStream);
+          return new SnappyCompressorInputStream(byteStream, 1 << 16 /* Avro uses 64KB blocks */);
```

Comment:
Instead of re-implementing the codec factory with a fixed number of codecs, why don't we just use the CodecFactory to create an instance of the codec which we can use to decode the bytes using Avro's code?